python/dgl/convert.py

```diff
@@ -386,10 +387,14 @@ def hetero_from_relations(rel_graphs, num_nodes_per_type=None):
         num_nodes_per_type = utils.toindex([num_nodes_per_type[ntype] for ntype in ntypes])
     ntype_dict = {ntype: i for i, ntype in enumerate(ntypes)}
     for rgrh in rel_graphs:
+        if rgrh._graph.dtype != index_dtype:
+            raise Exception("Expect relation graphs to be {}, but got {}".format(
+                index_dtype, rgrh._graph.dtype))
         stype, etype, dtype = rgrh.canonical_etypes[0]
         meta_edges_src.append(ntype_dict[stype])
         meta_edges_dst.append(ntype_dict[dtype])
         etypes.append(etype)
+    # TODO(Allen): metagraph is homograph, currently still using int64
```

Comment:
This PR didn't change `DGLGraph` part. Currently metagraph is a `DGLGraph` instead of a homo `DGLHeteroGraph`, and `DGLGraph` are all int64 graphs. Therefore currently here using int64. I think it's fine to keep it int64. How do you think about this?

Reply:
I see. I think as long as we document the behavior clearly (e.g., in the C++ class docstring), we are fine.